lib/Service/ModelService.php

```diff
@@ -78,22 +73,8 @@ public function modelFileExists(int $modelId, string $file): bool {
 	 * @return void
 	 */
 	public function prepareModelFolder(int $modelId) {
-		try {
-			$this->appData->getFolder('/models/' . $modelId);
-		} catch (NotFoundException $e) {
-			$this->appData->newFolder('/models/' . $modelId);
```

Comment:
I meant to revert your change, and just add that .. The creation of the models /MODEL_ID folder should be done here.

Reply:
What if I do
`public function prepareModelFolder(int $modelId) {
		if (!is_dir($this->modelsFolder)) {
			mkdir($this->modelsFolder . '/' . $modelID, 0770, true);
		}
	}
`

Reply:
> mkdir($this->modelsFolder . '/' . $modelID, 0770, true);

You are right you should use this in case of custom folder.. :sweat_smile: 

But in the default case, you should create the folder using the AppData methods,  because nextcloud sve this in the database and may fail.